src/txn.h

```diff
@@ -31,26 +36,30 @@ typedef enum {
 	TXN_ELEM_TXN_SIG,
 } txnElemType_e;
 
+typedef struct {
+    txnElemType_e elemType; // type of most-recently-seen element
+
+    uint8_t outVal[24];    // most-recently-seen currency value, Sia-encoded
```

Comment:
Encoded currency values should never need a length prefix larger than 1 byte, so in theory we could shave off another 7 bytes here. It would be slightly annoying since `cur2dec` expects an 8-byte prefix, but there are various ways of working around that. Would it be worth it, though? 🤔

Reply:
Probably.  It'd allow for a few more elements on the Nano S.